historical-adherence-query/java/Main.java

```diff
@@ -0,0 +1,203 @@
+            .withBasePath(region)
+            .build();
+
+        ApiResponse<AuthResponse> authResponse = apiClient.authorizeClientCredentials(clientId, clientSecret);
+
+        // Set the ApiClient instance as your default api client
+        Configuration.setDefaultApiClient(apiClient);
+
+        // Create an instance of UsersApi to retrieve your own userId
+        UsersApi usersApi = new UsersApi();
+        UserMe me = usersApi.getUsersMe(Collections.singletonList("presence"));
+        String myId = me.getId()
+
+        // Create an instance of the notification handler for your listeners
+        NotificationHandler notificationHandler = new NotificationHandler();
+        // Create a variable for your historical adherence event listener and instantiate it.
+        HistoricalAdherenceEventListener historicalAdherenceEventListener = new HistoricalAdherenceEventListener(myId);
+        // Subscribe to the listener
+        notificationHandler.addSubscription(historicalAdherenceEventListener);
+        // Connect to the web socket
+        notificationHandler.connect(true);
+
+        // Making an API call too soon after subscribing can result in the system thinking there are no subscribers.
+        TimeUnit.SECONDS.sleep(5);
+
+        String muId = "33333333-3333-3333-3333-333333333333"; // String | The management unit ID of the management unit
+
+        // Create a list of the users you want to query
+        List<String> userIds = new ArrayList<String>();
+        userIds.add("00000000-0000-0000-0000-000000000000");
+        userIds.add("11111111-1111-1111-1111-111111111111");
+        userIds.add("22222222-2222-2222-2222-222222222222");
+
+        // Create a calendar for setting the start and end date of the query
+        Calendar calendar = Calendar.getInstance();
+        calendar.set(Calendar.YEAR, 2020);
+        calendar.set(Calendar.MONTH, Calendar.JULY);
+        calendar.set(Calendar.DAY_OF_MONTH, 1);
+        calendar.set(Calendar.HOUR, 5);
+
+        Date startDate = calendar.getTime(); //Start date for the Historical Adherence query (2020-07-01T05:00:00.000Z)
+
+        // Now set the end date, here we are querying one day
+        calendar.set(Calendar.DAY_OF_MONTH, 2);
+        Date endDate = calendar.getTime(); //End date for the Historical Adherence query (2020-07-02T05:00:00.000Z)
+        String timeZone = "America/Chicago";
+
+        // Construct the body of the request using the variables set above
+        WfmHistoricalAdherenceQuery body = new WfmHistoricalAdherenceQuery(); // WfmHistoricalAdherenceQuery | body
+        body.setStartDate(startDate);
+        body.setEndDate(endDate);
+        body.setTimeZone(timeZone);
+        body.setUserIds(userIds);
+
+        // Create a Workforce Management Api Instance for the request
+        WorkforceManagementApi wfmApiInstance = new WorkforceManagementApi();
+        try
+        {
+            // Send the request using your workforce management api instance,
```

Comment:
Move these explanations into the tutorial's text. Comments in the code in a tutorial should just be a few words.